janus_core/calculations/md.py

```diff
@@ -442,6 +459,48 @@ def _restart_file(self) -> str:
             f"res-{step}.xyz", f"T{self.temp}", prefix_override=self.restart_stem
         )
 
+    def _parse_correlations(self):
+        """Parse correlation arguments into Correlations."""
+        if not self.correlation_kwargs:
+            self._correlations = None
+        elif len(self.correlation_kwargs["correlations"]) > 0:
+            if len(self.correlation_kwargs["correlation_parameters"]) != len(
+                self.correlation_kwargs["correlations"]
+            ):
+                raise ValueError("Mismatching sizes of correlations and parameters")
+
+            self._correlations = []
+            for cor in range(len(self.correlation_kwargs["correlations"])):
+                obs_a, obs_b = map(
+                    option_to_observable, self.correlation_kwargs["correlations"][cor]
+                )
+                params = self.correlation_kwargs["correlation_parameters"][cor]
+                self._correlations.append(Correlation(obs_a, obs_b, *params))
+
+    def _attach_correlations(self):
+        """Attach all correlations to self.dyn."""
+        # pylint: disable=W0640
```

Comment:
Not one I've come across before, but is this definitely not a problem?

If I do 

```
x = [1,2,3]
y = []
for i, _ in enumerate(x):
    y.append(lambda: i*2)
```
then `y[0]() == y[1]() == y[2]() == 2`, so in this case is it not similar? The final `i` would be the only one used in the `lambda` function?

Reply:
yes, I think so, ```functools.partial``` may do it?